oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/Expression.java

```diff
@@ -105,12 +106,11 @@ public SampleFamily propertyMissing(String metricName) {
                     return SampleFamily.EMPTY;
                 }
                 if (sampleFamilies.containsKey(metricName)) {
+                    ExpressionParsingContext.get().ifPresent(ctx -> ctx.samples.add(metricName));
                     return sampleFamilies.get(metricName);
                 }
-                if (!ExpressionParsingContext.get().isPresent()) {
-                    log.warn("{} referred by \"{}\" doesn't exist in {}", metricName, literal, sampleFamilies.keySet());
-                }
-                return SampleFamily.EMPTY;
+                // Using finder, support "."
+                return new SampleFamilyFinder(metricName, literal, sampleFamilies);
```

Comment:
Do you mean, `is there is no method found than failing back to a sample name including dot?`?

Reply:
If so, you are facing conflict with Prometheus and Telemetry rules.

Reply:
If there no property(not the method) found, it will append `"." + propertyName` to the meter name and trying to find.
If the method not found, it will same as the `SampleFamily`.

Reply:
Take a look how Prometheus metrics family did. All `.`s are replaced by `_` before the MAL engine.

Reply:
Replace all `.` to `_` in the Zabbix metric name.